compiler/src/dotty/tools/dotc/core/Contexts.scala

```diff
@@ -929,7 +929,10 @@ object Contexts {
             // - we don't want TyperState instantiating these TypeVars
             // - we don't want TypeComparer constraining these TypeVars
             val poly = PolyType(DepParamName.fresh(sym.name.toTypeName) :: Nil)(
-              pt => TypeBounds.empty :: Nil,
+              pt => (sym.info match {
+                case tb @ TypeBounds(lo, hi) if (lo eq defn.NothingType) && hi.isLambdaSub => tb
```

Comment:
The lower bound being Nothing or not shouldn't change anything here.

Reply:
My rationale was to restrict the change to just those bounds which only convey kind information. Happy to loosen that restriction if you think it's the right thing to do.

Reply:
If you don't want to loose the kind information when the lower bound is Nothing, then I don't think you want to loose it just because someone put a more precise lower bound. Try adding some extra testcases to clarify this.

Reply:
I agree that it should not matter whether lower bound is `Nothing`. Btw, to test for nothingness, use `lo.isRef(NothingClass)`. `eq` on types is unpredictable, since types might be re-generated in interactive scenarios between runs. 

I am also mystified by the treatment of the upper bound. As things stand, we make the upper bound `Any` for all first kinded types (including ones with a bound), but we copy the upper bound from `sym` if it is higher-kinded. That looks unsystematic to me. Does the precise bound matter at all? This is a question for @AleksanderBG .

Reply:
I've removed the restriction on the lower bound and added a couple more tests as suggested by @smarter. The last of these (with `List` as a lower bound) failed with the restriction in place.